src/components/ha-date-range-picker.ts

```diff
@@ -58,29 +62,32 @@ export class HaDateRangePicker extends LitElement {
         twentyfour-hours=${this._hour24format}
         start-date=${this.startDate}
         end-date=${this.endDate}
+        time-picker=${!this.hideTimePicker}
         ?ranges=${this.ranges !== undefined}
         first-day=${firstWeekdayIndex(this.hass.locale)}
       >
         <div slot="input" class="date-range-inputs">
           <ha-svg-icon .path=${mdiCalendar}></ha-svg-icon>
-          <ha-textfield
-            .value=${formatDateTime(this.startDate, this.hass.locale)}
-            .label=${this.hass.localize(
-              "ui.components.date-range-picker.start_date"
-            )}
-            .disabled=${this.disabled}
-            @click=${this._handleInputClick}
-            readonly
-          ></ha-textfield>
-          <ha-textfield
-            .value=${formatDateTime(this.endDate, this.hass.locale)}
-            .label=${this.hass.localize(
-              "ui.components.date-range-picker.end_date"
-            )}
-            .disabled=${this.disabled}
-            @click=${this._handleInputClick}
-            readonly
-          ></ha-textfield>
+          ${!this.minimal
+            ? html`<ha-textfield
+                  .value=${formatDateTime(this.startDate, this.hass.locale)}
+                  .label=${this.hass.localize(
```

Comment:
If the input is not displayed, we must use a `ha-button-icon` for accessibility.